src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1041,6 +1114,47 @@ internal override bool HasBackingField(SyntaxNode propertyOrIndexerDeclaration)
         internal override bool IsDeclarationWithInitializer(SyntaxNode declaration)
             => declaration is VariableDeclaratorSyntax { Initializer: not null } || declaration is PropertyDeclarationSyntax { Initializer: not null };
 
+        internal override bool IsRecordPrimaryConstructorParameter(SyntaxNode declaration)
+            => declaration is ParameterSyntax { Parent: ParameterListSyntax { Parent: RecordDeclarationSyntax } };
+
+        internal override bool IsRecordPrimaryConstructorProperty(SyntaxNode declaration)
+            => declaration is PropertyDeclarationSyntax { Parent: RecordDeclarationSyntax record, Identifier: { ValueText: var name } }
+            && record.ParameterList is not null
+            && record.ParameterList.Parameters.Any(p => p.Identifier.ValueText.Equals(name));
```

Comment:
This needs to use symbols, or it won't work with partial types.

Reply:
Still an issue?

---
In reply to: [604551031](https://github.com/dotnet/roslyn/pull/51926#discussion_r604551031) [](ancestors = 604551031)

Reply:
I think I commented elsewhere, but no, this works with partial types because `record.ParameterList` resolves to the right thing, regardless of which partial the parameters are defined in. I added a test that covers it just in case.

Reply:
I don't see how it can work. Which test is it?

---
In reply to: [611284259](https://github.com/dotnet/roslyn/pull/51926#discussion_r611284259) [](ancestors = 611284259)

Reply:
Nevermind, I was wrong. The test I had for this was incorrect.